app/src/server/api/routers/users.router.ts

```diff
@@ -120,7 +138,7 @@ export const usersRouter = createTRPCRouter({
         },
       });
 
-      if (!invitation) {
+      if (!invitation || invitation.isCanceled) {
```

Comment:
I would add isCanceled = false to the database queries. Then you can get rid of this check after the fact.